test/CMakeLists.txt

```diff
@@ -54,35 +56,27 @@ else()
 endif()
 
 
-
-
-
-
 ### Download and unpack googletest at configure time
 # (from the googletest docs)
-configure_file(CMakeLists.txt.in googletest-download/CMakeLists.txt)
-execute_process(COMMAND ${CMAKE_COMMAND} -G "${CMAKE_GENERATOR}" .
-  RESULT_VARIABLE result
-  WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/googletest-download )
-if(result)
-  message(FATAL_ERROR "CMake step for googletest failed: ${result}")
-endif()
-execute_process(COMMAND ${CMAKE_COMMAND} --build .
-  RESULT_VARIABLE result
-  WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/googletest-download )
-if(result)
-  message(FATAL_ERROR "Build step for googletest failed: ${result}")
+if(${CMAKE_VERSION} VERSION_LESS 3.14)
```

Comment:
I notice that this tests against version 3.14 here, but 3.11 in `deps/CMakeLists.txt`. Is the difference intentional?

Reply:
**This difference is intentional!** 
One of my frustrations with CMake is that the API is a moving target. Theres a call to `FetchContent_MakeAvailable()` which was introduced in 3.14. The file/module defining these other calls are copied verbatim from some future CMake so we can use this function. When CMake 3.14 is commonplace then it would make sense to use FetchContent definitions from the CMake install.

**Why does FetchContent for Eigen only require 3.11?** 
That's when FetchContent was introduced in general. Here since `FetchContent_MakeAvailable` will build Eigen tests and other things which are undesirable, we're only using the `Populate` and other calls which defined in the original FetchContent module. Actually the following code to populate etc is very similar to that in the contents of `MakeAvailable`. We just remove the part to recurse into the subdirectory. We substitute in some logic to check the Eigen headers and define the relevant target which is the minimal thing to do.

There is one potential motivation to recurse into the Eigen subdirectory. Eigen CMakeLists.txt defines a bunch of compiler tests and flag definitions which may improve the performance of a subsequent build. I haven't looked closely at this and can't comment at this point on if it will actually matter or not.